tests/third_party/cupy/linalg_tests/test_decomposition.py

```diff
@@ -0,0 +1,248 @@
+import unittest
+
+import numpy
+import pytest
+
+import dpnp as cupy
+from tests.helper import has_support_aspect64
+from tests.third_party.cupy import testing
+from tests.third_party.cupy.testing import condition
+
+
+def stacked_identity(xp, batch_shape, n, dtype):
+    shape = batch_shape + (n, n)
+    idx = xp.arange(n)
+    x = xp.zeros(shape, dtype=dtype)
+    x[..., idx, idx] = 1
+    return x
+
+
+@testing.parameterize(
+    *testing.product(
+        {
+            "full_matrices": [True, False],
+        }
+    )
+)
+# @testing.fix_random()
+class TestSVD(unittest.TestCase):
+    def setUp(self):
+        self.seed = numpy.random.randint(0x7FFFFFFF)
+
+    @testing.for_dtypes(
+        [
+            numpy.int32,
+            numpy.int64,
+            numpy.uint32,
+            numpy.uint64,
+            numpy.float32,
+            numpy.float64,
+            numpy.complex64,
+            numpy.complex128,
+        ]
+    )
+    def check_usv(self, shape, dtype):
+        array = testing.shaped_random(shape, numpy, dtype=dtype, seed=self.seed)
+        a_cpu = numpy.asarray(array, dtype=dtype)
+        a_gpu = cupy.asarray(array, dtype=dtype)
+        result_cpu = numpy.linalg.svd(a_cpu, full_matrices=self.full_matrices)
+        result_gpu = cupy.linalg.svd(a_gpu, full_matrices=self.full_matrices)
+        # Check if the input matrix is not broken
+        testing.assert_allclose(a_gpu, a_cpu)
+
+        assert len(result_gpu) == 3
+        for i in range(3):
+            assert result_gpu[i].shape == result_cpu[i].shape
+            if has_support_aspect64():
+                assert result_gpu[i].dtype == result_cpu[i].dtype
+            else:
+                assert result_gpu[i].dtype.kind == result_cpu[i].dtype.kind
+        u_cpu, s_cpu, vh_cpu = result_cpu
+        u_gpu, s_gpu, vh_gpu = result_gpu
+        testing.assert_allclose(s_gpu, s_cpu, rtol=1e-5, atol=1e-4)
+
+        # reconstruct the matrix
+        k = s_cpu.shape[-1]
+
+        # dpnp.dot/matmul does not support complex type and unstable on cpu
+        # TODO: remove it when dpnp.dot/matmul is updated
+        u_gpu = cupy.asnumpy(u_gpu)
+        vh_gpu = cupy.asnumpy(vh_gpu)
+        s_gpu = cupy.asnumpy(s_gpu)
+        xp = numpy
+
+        if len(shape) == 2:
+            if self.full_matrices:
+                a_gpu_usv = xp.dot(u_gpu[:, :k] * s_gpu, vh_gpu[:k, :])
```

Comment:
here and after we can explicitly call numpy
```suggestion
                a_gpu_usv = numpy.dot(u_gpu[:, :k] * s_gpu, vh_gpu[:k, :])
```